dali/python/nvidia/dali/ops.py

```diff
@@ -404,21 +404,30 @@ def __init__(self, **kwargs):
             self._spec.AddArg("preserve", self._preserve)
             self._preserve = self._preserve or self._schema.IsNoPrune()
 
-            if "output_dtype" in kwargs.keys():
-                if "dtype" in kwargs.keys():
-                    raise TypeError(
-                        ("Operator {} got an unexpected 'output_dtype' argument when " +
-                         "'dtype' is already provided").format(type(self).__name__))
-                # show only this warning
+            # Check for any deprecated arguments that should be replaced or removed
+            arg_names = list(kwargs.keys())
+            for arg_name in arg_names:
+                if not self._schema.IsDeprecatedArg(arg_name):
+                    continue
+                meta = self._schema.DeprecatedArgMeta(arg_name)
+                new_name = meta['renamed_to']
+                removed = meta['removed']
+                msg = meta['msg']
+                if new_name:
+                    if new_name in kwargs:
+                        raise TypeError("Operator {} got an unexpected 'output_dtype' argument when 'dtype' is already provided".format(type(self).__name__))
+                    if msg:
+                        msg += " " if msg.endswith('.') else ". "
+                    msg += "Use \'{}\' instead.".format(new_name)
+                    kwargs[new_name] = kwargs[arg_name]
+                    del kwargs[arg_name]
+                elif removed:
+                    msg += " The argument is no longer used and should be removed."
```

Comment:
Same for this. I think it would be better if we can see the deprecation message being constructed in one place.